lib/puppet/functions/simplib/passgen.rb

```diff
@@ -83,18 +83,21 @@ def passgen(identifier, modifier_hash=nil)
     unless File.directory?(settings['keydir'])
       begin
         FileUtils.mkdir_p(settings['keydir'],{:mode => 0750})
-        # This chown is applicable as long as it is applied
-        # by puppet, not puppetserver.
-        FileUtils.chown(settings['puppet_user'],
-         settings['puppet_group'], settings['keydir']
-       )
       rescue SystemCallError => e
         err_msg = "simplib::passgen: Could not make directory" +
          " #{settings['keydir']}:  #{e.message}. Ensure that" +
          " #{File.dirname(settings['keydir'])} is writable by" +
          " '#{settings['puppet_user']}'"
         fail(err_msg)
       end
+      begin
```

Comment:
Line 10 has a comment about failure related to puppet permissions that is not longer correct.